src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1223,15 +1074,9 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
                 case TYP_UBYTE:
                 case TYP_INT:
                 case TYP_UINT:
-                    if (!compExactlyDependsOn(InstructionSet_SSE41))
-                    {
-                        return nullptr;
-                    }
-                    break;
-
                 case TYP_LONG:
                 case TYP_ULONG:
-                    if (!compExactlyDependsOn(InstructionSet_SSE41_X64))
+                    if (!compExactlyDependsOn(InstructionSet_SSE41))
```

Comment:
> What happens here when we run on X64?

We will decompose into two 32-bit extractions instead.

> Is this a reason why the intrinsic was dependent previously on InstructionSet_SSE41_X64 ISA

Because we were only using `Sse41.X64.Extract` and hadn't enabled the same handling/optimizations that were available for the `Vector<T>.get_Item` scenario.